reconcile/utils/saasherder/saasherder.py

```diff
@@ -1156,11 +1169,19 @@ def _get_and_validate_image(
                 logging.error(
                     f"{error_prefix} Image : {full_image_path} does not exist"
                 )
+        # catches all exceptions
+        # if the exception contains 'authentication', log a different message to help debugging
         except Exception as e:
-            logging.error(
-                f"{error_prefix} Image is invalid: {full_image_path}. "
-                + f"details: {e!s}"
-            )
+            if 'authentication' in str(e).lower():
+                logging.error(
+                    f"{error_prefix} Failed to authenticate to the repository for image: {full_image_path}. "
+                    + f"details: {e!s}"
+                )
+            else:
+                logging.error(
+                    f"{error_prefix} Image is invalid: {full_image_path}. "
+                    + f"details: {e!s}"
+                    )
```

Comment:
if "authentication" is under the details in `f"details: {e!s}"`, why do we need more info?

Reply:
The `details` were previously included with the error messages, so I kept it there for consistency.

Reply:
the old message print out is `does not exist`, which means this error handling it not executed, so changes here won't work. If this blocked executed, then we should see `details: authentication xxx`

The root issue is here https://github.com/app-sre/sretoolbox/blob/dacd722f555f0315aab881621f6118b66e39f3bc/sretoolbox/container/image.py#L597-L601

401/403 error already handled and return False, we should check when image not exist, what response should be returned (404?) and only handle that case, re-throw other exceptions